lib/puppet/network/http/factory.rb

```diff
@@ -23,10 +23,13 @@ def create_connection(site)
     Puppet.debug("Creating new connection for #{site}")
 
     args = [site.host, site.port]
-    if Puppet[:http_proxy_host] == "none"
-      args << nil << nil
-    else
-      args << Puppet[:http_proxy_host] << Puppet[:http_proxy_port]
+
+    unless Puppet::Util::HttpProxy.no_proxy?(site)
```

Comment:
Same as for pkgdmg - there are existing tests for proxy management here, it would be cool to see them expanded to cover the no proxy case.